packages/protocol/contracts/common/ProxyCloneFactory.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.5.13;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+import "./InitializableProxy.sol";
+
+contract ProxyCloneFactory is Ownable {
+  address public proxyAddress;
```

Comment:
Potential slight optimization: do we need the `proxyAddress` here to be variable? Could set it as a constant to avoid a storage read and bytecode splicing costs in each deployment.

The factory is stateless, so if we want to start using a different Proxy we can just deploy a fresh Factory, right?

Reply:
Additionally, gas costs could be reduced even further by deploying the InitializableProxy to an address with several leading 0 bytes: https://github.com/optionality/clone-factory#use-vanity-contract-addresses-for-even-cheaper-clone-contracts

Reply:
Correct we could remove this, and we could use vanity addresses.

However, removing it makes it difficult to test (the address varies per-env), and saves very little on gas. IMO, more trouble than it's worth.

Similarly, the vanity address saves very little in gas and IMO is also more trouble than it's worth.